### PR TITLE
Default RG and ACR Values with Publisher Name

### PR DESCRIPTION
**Pull request by jordlay**
- Added default rg and acr values for nf + nsd based on publisher name
- Added blob_url default for VNF based on publisher name
- Updated helptext 
- Added fix for bug https://dev.azure.com/msazuredev/AzureForOperators/_workitems/edit/920284, as very minor change

Testing when no RG or ACR given:
- VNF no errors on build (did not try to test further)
- CNF no errors on build, successful publish of publisher, RG and ACR
- NSD: no errors on build, successful publish of all resources

Testing with both RG and ACR given:
- Tested with CNF and NSD, behaviour as before
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/aosm/azext_aosm/_configuration.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def validate(self):
         Validate the configuration.
         """
         if not self.version:
-            raise ValidationError("version must be set.")
+            raise ValidationError("Version must be set.")
```
Comment (jamiedparsons) [inline src/aosm/azext_aosm/_configuration.py:60]: Minor thing, but I've been trying to keep the error messages referencing fields exactly as they have to be supplied in input.json.  So I think that lowercase v is the right thing here.

File: src/aosm/azext_aosm/_configuration.py [hunk 2]
```diff
@@ -383,6 +398,7 @@ def __post_init__(self):
 
         Used when creating CNFConfiguration object from a loaded json config file.
         """
+        super().__post_init__()
```
Comment (jamiedparsons) [inline src/aosm/azext_aosm/_configuration.py:401]: Good spot on all these super() calls, I'd have forgotten to do that!

File: src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py [hunk 3]
```diff
@@ -709,12 +709,12 @@ def _replace_values_with_deploy_params(
                         final_values_mapping_dict[k].append(
                             self._replace_values_with_deploy_params(item, param_name)
                         )
-                    elif isinstance(v, (str, int, bool)) or not v:
+                    elif isinstance(item, (str, int, bool)) or not v:
```
Comment (jamiedparsons) [inline src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py:712]: This is a fix for another bug - is it meant to be in this PR?

**Review by jamiedparsons — COMMENTED** [review]
Looks good a couple of very minor comments.

File: src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py (hunk 3 above)
Comment (jordlay) [reply src/aosm/azext_aosm/generate_nfd/cnf_nfd_generator.py:712]: Added it in here so it could just be fixed asap. I'll add it to the PR description

**Pull request merged**